modules/steam/autostart.nix

```diff
@@ -31,7 +31,15 @@ in
         };
 
         desktopSession = mkOption {
-          type = types.nullOr types.str;
+          type = with types ; nullOr str // {         
+            check = userProvidedDesktopSession:
+              lib.assertMsg  (str.check userProvidedDesktopSession && userProvidedDesktopSession ? config.services.xserver.displayManager.sessionData.sessionNames) ''
```

Comment:
Should probably check if `userProvidedDesktopSession` is null first.

Also, nit: unnecessary extra space after `assertMsg`.